src/partition/repart.c

```diff
@@ -3929,8 +3939,8 @@ static int partition_format_verity_hash(
                                 .flags = CRYPT_VERITY_CREATE_HASH,
                                 .hash_name = "sha256",
                                 .hash_type = 1,
-                                .data_block_size = context->sector_size,
-                                .hash_block_size = context->sector_size,
+                                .data_block_size = arg_sector_size ? context->sector_size : 4096,
+                                .hash_block_size = arg_sector_size ? context->sector_size : 4096,
```

Comment:
we have this rule that we only do downgrade-to-bool zero checks for stuff that is a boolean or boolean-like. For other stuff, i.e. arbitrary integer checks we do explicit zero checks, i.e. `arg_sector_size != 0 ? …`